### DISCUSSION TITLE
Theoratically can't Inko be faster than rust?

### DISCUSSION THREAD

**Discussion by NoelJacob** [inko-lang/inko#659] [General]
Maybe if we can do something about Inko's Rust runtime.

**Comment by yorickpeterse** [comment]
Due to Inko's nature (reference counting for references, heap allocating by default, etc), I suspect it won't be able to outperform Rust as a whole. This isn't a goal either, as Inko isn't a systems language meant to give you complete control over everything. The Rust runtime isn't really the issue either, it's mostly used for scheduling processes and providing some helper functions used through the FFI.

I'm hoping performance one day will be comparable to languages like Go, Java, etc: basically good enough for 90% of the cases.
